### DISCUSSION TITLE
直接配置为客户端的网关和DNS或者客户端设置路由可以访问，但主路由设置静态路由无法访问

### DISCUSSION THREAD

**Discussion by wally94** [kkkgo/PaoPaoGateWay#137] [问和答] [answered]
使用的TPLINK的poe一体机，有静态路由的设置，直接设置为网关可以使用，使用静态路由无法连接，这是什么原因？
![微信截图_20241018104431](https://github.com/user-attachments/assets/d079ef6d-9624-404d-bd9f-91caed0c6a0e)
配置为：
#paopao-gateway
mode=suburl
fake_cidr=198.18.0.0/16
dns_ip=10.0.1.2
dns_port=53
clash_web_port="80"
clash_web_password="clashpass"
openport=no
udp_enable=yes
sleeptime=30
fast_node=yes
test_node_url="https://www.youtube.com/generate_204"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"
fall_direct="no"
dns_burn=no
ex_dns="223.5.5.5:53"
net_rec=no
max_rec=5000
suburl="https://v1.v2ai.top/link/yxxxxxxx"
subtime=1d

**Comment by kkkgo** [comment]
你应该跟踪一下路由看看静态路由是否可达。比如tracert google.com

**Reply by wally94** [reply]
![微信截图_20241018105930](https://github.com/user-attachments/assets/9fc84464-95cc-4f9f-aaf0-6c0f3236745b)

**Reply by kkkgo** [reply]
既然路由可达，你就可以用curl命令行看看能不能访问网站，比如 curl http://google.com

**Reply by wally94** [reply]
![微信截图_20241018111230](https://github.com/user-attachments/assets/5f361b12-ba98-408c-be32-0e7373de7220)
手动在pc添加一条路由后访问也没有问题，是主路由的问题吗？

**Reply by kkkgo** [reply]
也没有问题是指什么？所以你curl到底能不能访问？

**Reply by wally94** [reply]
![微信截图_20241018111601](https://github.com/user-attachments/assets/993067d4-2a3a-49ef-bce6-813b7b302bf6)
不能访问

**Reply by kkkgo** [reply]
所以你手动添加本地路由后能访问吗？

**Reply by wally94** [reply]
手动添加后可以访问

**Reply by kkkgo** [reply, marked as answer]
那就是静态路由不可达，部分家用路由器产品交换端口存在静态路由bug可能会导致路由不可达，特别是当AP交换机用的情况。

**Reply by wally94** [reply]
这样啊，我折腾了好久好久各种方法都试过了，看来我得换主路由了，感谢感谢

**Reply by wally94** [reply]
已经测试了，接在主路由的网口上，没有问题，正常使用，接在AP的网口上无法访问，看来确实是AP的问题

**Comment by LetMeDecay** [comment]
我主路由xdr6088刷的openwrt，电脑直插能路由到gateway去
但是只要接了xdr5480当ap，再接电脑，电脑就路由不过去了
tplink真是💩

~正常情况tracert应该是先跳到主路由ip，然后再跳到fakeip去，插上ap端口就直接跳到fakeip去了~
翻了下回复，发现跟你上面情况一模一样，还是先临时用几天吧，等后面交换机回来看看有没有问题

update:
研究了一阵子发现了这个issue，https://github.com/istoreos/istoreos/issues/596
然后我试着打开了主路由openwrt防火墙lan上的ip动态伪装，结果就搞定了，op和其他有同样问题的可以试试这个
缺点就是主路由会再进行一次nat，同时外网进来的ip全部都变成主路由ip了

**Reply by kkkgo** [reply]
讲个小故事，xdr5480我真买过……
之前看评测说信号好寻思拿来当AP，不过实际上挺烫的，信号跟6088也没差多少吧，但比6088烫。
这不是重点，xdr5480我是和tp的企业路由器一起买的，当时调出了问题，找了tp的企业技术支持，排查了半天，抓包，都没有发现问题，最后还是我调试发现是xdr5480的问题。
然后给我转家用产品技术支持，又给我排查测试抓包半天，一开始还不承认，后来他自己也重现了，问怎么解决，没有解决方案。
这两位tp工程师还躺在我的微信列表里。
这些都不重要，最重要的是，xdr5480是没有任何第三方固件可以刷的，闪存特别小不说，CPU系列来说也很小众。
所以，以后我都不买不能刷机的家用路由器……（当然，企业级AP除外）

**Reply by LetMeDecay** [reply]
😭在让人失望这方面tp从来都不让人失望，我怀疑是不是这吊毛固件把主路由的icmp报文给吞了
买5480 v1就是为了搞个便宜货里消费级最强信号，结果还是吃了大亏
这玩意还是后面出掉算了，关键是手机走连wifi就会触发，再加上同信号漫游就更蛋疼了
